flatflow/scheduler/internal/algorithm/regression_test.cc

```diff
@@ -0,0 +1,123 @@
+// Copyright 2024 The FlatFlow Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "flatflow/scheduler/internal/algorithm/regression.h"
+
+#include <random>
+#include <vector>
+
+#include "absl/base/log_severity.h"
+#include "absl/log/globals.h"
+#include "absl/log/initialize.h"
+#include "absl/log/internal/globals.h"
+#include "absl/log/log.h"
+#include "gtest/gtest.h"
+
+namespace {
+
+template <int degree>
+class Regression : public flatflow::scheduler::internal::algorithm::
+                       PassiveAggressiveRegressor<degree> {
+ public:
+  Regression(const double epsilon)
+      : flatflow::scheduler::internal::algorithm::PassiveAggressiveRegressor<
+            degree>(epsilon) {}
+  void fit(const std::vector<double> &X, const std::vector<double> &y) {
+    flatflow::scheduler::internal::algorithm::PassiveAggressiveRegressor<
+        degree>::fit(X, y);
+  }
+
+  inline std::vector<double> &getCoef() { return this->coef_; }
+};
+
+class RegressionTest : public testing::Test {
+ protected:
+  void SetUp() override {
+    if (!absl::log_internal::IsInitialized()) {
+      absl::InitializeLog();
+      absl::SetStderrThreshold(absl::LogSeverity::kInfo);
+    }
+    std::random_device rd;
+    std::mt19937 gen(rd());
+    std::uniform_real_distribution<double> dis(100.0, 120.0);
+    for (std::size_t idx = 0; idx < kDatasetSize; ++idx) {
+      X[idx] = dis(gen);
+    }
+  }
+
+  double generateDummy(double x, std::vector<double> coef) {
+    double sum = 0;
+    for (std::size_t coefIdx = 0; coefIdx < coef.size(); ++coefIdx) {
+      sum += coef[coefIdx] * std::pow(x, coefIdx);
+    }
+    return sum;
+  }
+
+  std::vector<double> generateCoefficients(std::size_t degree) {
+    std::vector<double> coeffs;
+    std::random_device rd;
+    std::mt19937 gen(rd());
+    std::uniform_real_distribution<double> dis(300.0, 400.0);
+    for (std::size_t coefIdx = 0; coefIdx <= degree; ++coefIdx) {
+      coeffs.push_back(dis(gen));
+    }
+    return coeffs;
+  }
+
+  void print(const std::vector<double> &modelCofficients,
```

Comment:
I think ``print`` is not feasible for this function name. ``print()`` does not print anything!